shell/components/ResourceDetail/Masthead.vue

```diff
@@ -370,7 +376,7 @@ export default {
     },
 
     hideNamespaceLocation() {
-      return this.$store.getters['currentProduct'].hideNamespaceLocation;
+      return this.$store.getters['currentProduct'].hideNamespaceLocation || !this.$store.getters['management/byId'](MANAGEMENT.CLUSTER, LOCAL_CLUSTER);
```

Comment:
This is implementation specific code in a generic component. In prov cluster model we could overload `namespaceLocation` to return false if the user cannot see the local cluster ... and then here check for value.namespaceLocation === false

Reply:
Yeah, `=== BLANK_CLUSTER ? this.value?.localCluster?.id : this.$route.params.cluster` is a typo from first commit, it can be removed.
I would set `namespaceLocation` to `null` in case the local cluster is not visible.

Reply:
Unfortunately it needs to be something specific rather than a falsey. Lots of models don't have a `namespaceLocation` which means the ns link will be hidden (instead of falling back on the generated one in)

```
namespaceLocation() {
      if (!this.isNamespace) {
        return this.value.namespaceLocation || {
          name:   'c-cluster-product-resource-id',
```

Reply:
I think it could be simply `this.value.namespaceLocation === null`. In the other models it will be undefined.